cfme/containers/provider/__init__.py

```diff
@@ -601,24 +540,15 @@ def remove_label(self, name, silent_failure=False):
         Raises:
             :py:class:`LabelNotFoundException`.
         """
-        json_content = self._get_json()
-        if name not in json_content['metadata'].get('labels', {}).keys():
-            failure_signature = 'Could not find label "{}", labels: {}' \
-                .format(name, json_content['metadata']['labels'])
+        try:
+            self.mgmt.delete_label(name)
+            return True
+        except:
```

Comment:
**REQUIRED**
It should be at least `except Exception:` but it's better to catch the exact exception.